py/datastore_service.py

```diff
@@ -85,15 +58,16 @@ def make_thread_locals(self):
       if cursor.fetchone()[0] == False:
         try:
           prepare_statement = "PREPARE report AS INSERT INTO segments (segment_id,prev_segment_id,mode," \
-                              "start_time,end_time,length,provider) VALUES ($1,$2,$3,$4,$5,$6,$7);"
+                              "start_time,start_time_dow,start_time_hour,end_time,length,speed,provider) " \
```

Comment:
why dont you n eed end time dow and hour? a reading could surely cross from one into another. isn't this the no one routes at midnight bug?